File: app/src/main/java/pers/victor/androiddemos/activity/VerticalMenuActivity.java
package pers.victor.androiddemos.activity;

import android.view.View;

import pers.victor.androiddemos.R;
import pers.victor.androiddemos.util.ShowToast;
import pers.victor.androiddemos.view.VerticalMenuLayout;

/**
 * Created by Victor on 2017/4/1. (ง •̀_•́)ง
 */

public class VerticalMenuActivity extends ToolbarActivity {
    @Override
    public int bindLayout() {
        return R.layout.activity_vertical_menu;
    }

    @Override
    public void initView() {
        VerticalMenuLayout verticalMenuLayout = $(R.id.vertical_menu_layout);
        for (int i = 0; i < verticalMenuLayout.getChildCount(); i++) {
            final int finalI = i;
            verticalMenuLayout.getChildAt(i).setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    ShowToast.shortToast(finalI);
                }
            });
        }
    }
}

File: app/src/main/kotlin/pers/victor/androiddemos/activity/VerticalMenuActivity.kt
package pers.victor.androiddemos.activity

import pers.victor.androiddemos.R
import pers.victor.androiddemos.util.ShowToast
import kotlinx.android.synthetic.main.activity_vertical_menu.*

/**
 * Created by victor on 16-8-1.
 */
class VerticalMenuActivity : ToolbarActivity() {
    override fun bindLayout() = R.layout.activity_vertical_menu

    override fun initView() {
        for (i in 0..vertical_menu_layout.childCount - 1) {
            vertical_menu_layout.getChildAt(i).setOnClickListener {
                ShowToast.shortToast("第 $i 个")
            }
        }
    }

}